docs/blockentities/container.md

```diff
@@ -0,0 +1,302 @@
+        return direction == Direction.UP && index == 0;
+    }
+}
+```
+
+## Using Containers
+
+Now that we have created containers, let's use them!
+
+Since there is a considerable overlap between `Container`s and `BlockEntity`s, containers are best retrieved by casting the block entity to `Container` if possible:
+
+```java
+if (blockEntity instanceof Container container) {
+    // do something with the container
+}
+```
+
+The container can then use the methods we mentioned before, for example:
+
+```java
+// Get the first item in the container.
+ItemStack stack = container.getItem(0);
+
+// Set the first item in the container to dirt.
+container.setItem(0, new ItemStack(Items.DIRT));
+
+// Removes a quantity of (up to) 16 from the third slot.
+container.removeItem(2, 16);
+```
+
+:::warning
+A container may throw an exception if trying to access a slot that is beyond its container size. Alternatively, they may return `ItemStack.EMPTY`, as is the case with (for example) `SimpleContainer`.
+:::
+
+## `Container`s on `ItemStack`s
+
+Until now, we mainly discussed `Container`s on `BlockEntity`s. However, they can also be applied to [`ItemStack`s][itemstack] using the `minecraft:container` [data component][datacomponent]:
+
+```java
+// We use SimpleContainer as the superclass here so we don't have to reimplement the item handling logic ourselves.
+// Due to implementation details of SimpleContainer, this may lead to race conditions if multiple parties
+// can access the container at the same time, so we're just going to assume our mod doesn't allow that.
+// You may of course use a different implementation of Container (or implement Container yourself) if needed.
+public class MyBackpackContainer extends SimpleContainer {
+    // The item stack this container is for. Passed into and set in the constructor.
+    private final ItemStack stack;
+    
+    public MyBackpackContainer(ItemStack stack) {
+        // We call super with our desired container size.
+        super(27);
+        // Setting the stack field.
+        this.stack = stack;
+        // We load the container contents from the data component (if present), which is represented
+        // by the ItemContainerContents class. If absent, we use ItemContainerContents.EMPTY.
+        ItemContainerContents contents = stack.getOrDefault(DataComponents.CONTAINER, ItemContainerContents.EMPTY);
+        // Copy the data component contents into our item stack list.
+        contents.copyInto(this.getItems());
+    }
+
+    // When the contents are changed, we save the data component on the stack.
+    @Override
+    public void setChanged() {
+        super.setChanged();
+        this.stack.set(DataComponents.CONTAINER, ItemContainerContents.fromItems(this.getItems()));
+    }
+}
+```
+
+And voilà, you have created an item-backed container! Simply call `new MyBackpackContainer(stack)` to create a container for a menu or other use case.
```

Comment:
This should note that wrapping a data component in a `Container` for use in a menu requires the use of a container-compatible implementation of NeoForge's `StackCopySlot`, otherwise the menu will mutate the data component, breaking the immutability contract of data components.